exist-versioning-release.md

```diff
@@ -93,40 +93,121 @@ That version number is also copied into the `META-INF/MANIFEST.MF` file of any J
+This section details concrete steps for creating and publishing product releases. 
 
-### Initiating Semantic Versioning
+### Release preprations and prerequisites 
 
-Version 3.0.0 was released before Semantic Versioning. The following steps will initiate Semantic Versioning for the remainder of the development phase of the next release, a new minor version to be called version 3.1.0:
+#### Required Software 
+Once development on a new stable version is complete, the following steps will prepare the version for release. For purposes of illustration, we will assume we are preparing the stable release of version 5.3.0.
+You will require a system with:
+
+* macOS
+* JDK 8
+* Maven 3.6.0+
+* Docker
+* GnuPG
+
+#### Required Keys, Credentials and Permissions
+
+##### GPG Key / Keystore and Credentials
+A private GPG key for signing release artifacts is required. To gather the required information run `gpg --list-secret-keys --keyid-format SHORT`
+
+```
+gpg --list-secret-keys --keyid-format SHORT
+/Users/username/.gnupg/pubring.kbx
+----------------------------------
+sec   rsa4096/12345678 2013-04-26 [SC]
+```
+
+This gives you the keystore you are using (here: `/Users/username/.gnupg/pubring.kbx`) and your key-id (here: `12345678` the part after rsa4096/). 
+
+**Important Note** 
+GnuPG is using an [optimized keystore since version 2.1](https://www.gnupg.org/documentation/manuals/gnupg/GPG-Configuration.html) which contains public and secret keys. Former GnuPG version had seperated keystore for public (~/.gnupg/pubring.gpg) and private keys (~/.gnupg/secring.gpg). 
+
+###### Maven Keystore values prior GnuPG 2.1
+
+* existdb.release.public-keyfile=~/.gnupg/pubring.gpg
+* existdb.release.private-keyfile=~/.gnupg/secring.gpg
+
+###### Maven Keystore values for GnuPG 2.1 or newer
+
+* existdb.release.public-keyfile=~/.gnupg/pubring.kbx
+* existdb.release.private-keyfile=~/.gnupg/pubring.kbx
+                   
+To test gpg simply create a file `test.txt` and run`gpg -ab test.txt`to sign it. you should see a new file `test.txt.asc` which means GnuPG signing worked fine. 
+
+##### Java KeyStore 
+* To create an eXist-db release a Java KeyStore with key is needed to sign the IzPack Installer. In case you don't have a Java KeyStore you can create one with: 
+
+```bash
+keytool -genkey \
+  -alias some-alias \
+  -keystore keystore.p12 \
+  -storetype PKCS12 \
+  -keyalg RSA \
+  -storepass some-pass \
+  -validity 7300 \
+  -keysize 4096
+```
+
+To verify if your Java KeyStore works fine run 
+
+```bash
+keytool -list -v -keystore /path/to/keystore.p12
+```
 
-1.  Modify `$EXIST_HOME/build.properties` to read:
+##### Apple Developer Certificate
 
+A valid Apple Developer Certificate to sign Mac DMG is required. If you don't have one you will need to join the [Apple Developer Program](https://developer.apple.com/programs/enroll/) and to create a certificate via XCode. 
```

Comment:
`to sign Mac DMG is` -> `to sign both the Mac App and Mac DMG is`

Reply:
As I understand it the .app is created in `mvn release:prepare` whereas the .dmg is an artifact created in `mvn release:perform`.